### PR TITLE
feat(grant): add filter by identifier

### LINKED ISSUE #1505: Filter grants by payment pointer (identifier)
For grant management user should see only his own grants.
List grants graphql endpoint should allow filtering by identifier (payment pointer)

### PR DESCRIPTION
**Pull request by dragosp1011**
## Changes proposed in this pull request

- Add filter of grants by identifier defined in grant access. (As identifier does not exist on grant itself in db)
- Add graphql query to get grant by id

## Context

fixes #1505 
## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/auth/src/graphql/schema.graphql [hunk 1]
```diff
@@ -40,6 +43,14 @@ input PaginationInput {
   last: Int
 }
 
+input GrantFilter {
+  identifier: FilterString
```
Comment (mkurapov) [inline packages/auth/src/graphql/schema.graphql:47]: @sabineschaller what do you think about always requiring `identifier` in any Open Payments Grant Request? Does it make sense to allow issuing grants for any payment pointer?

File: packages/auth/src/graphql/schema.graphql [hunk 2]
```diff
@@ -1,6 +1,9 @@
 type Query {
   "Fetch a page of grants."
-  grants(input: PaginationInput): GrantsConnection!
+  grants(input: PaginationInput, filter: GrantFilter): GrantsConnection!
```
Comment (mkurapov) [inline packages/auth/src/graphql/schema.graphql:3]: @BlairCurrey what do you think about the `input: PaginationInput` object? vs like you have for the webhook events:

https://github.com/interledger/rafiki/blob/b96ddf1a865143a5c87b3705594b3cd5c2f48287/packages/backend/src/graphql/schema.graphql#L57-L68

**Review by mkurapov — COMMENTED** [review]
Looks good, just want to get some thoughts from Blair & Sabine

File: packages/auth/src/graphql/schema.graphql (hunk 2 above)
Comment (BlairCurrey) [reply packages/auth/src/graphql/schema.graphql:3]: Although my initial instinct was a pagination input like what we have in the `auth` package, we settled on separate `after`, `before`, `first`, `last` args for consistency with the relay spec and conventions around implementing it. I think that's probably still the right approach and ideally we standardize across packages.

File: packages/auth/src/graphql/resolvers/grant.ts [hunk 3]
```diff
@@ -17,7 +17,8 @@ export const getGrants: QueryResolvers<ApolloContext>['grants'] = async (
   ctx
 ): Promise<ResolversTypes['GrantsConnection']> => {
   const grantService = await ctx.container.use('grantService')
-  const grants = await grantService.getPage(args.input)
+  const { filter, input: pagination } = args
+  const grants = await grantService.getPage(pagination, filter)
   const pageInfo = await getPageInfo(
     (pagination: Pagination) => grantService.getPage(pagination),
```
Comment (BlairCurrey) [inline packages/auth/src/graphql/resolvers/grant.ts:23]: I think the `pageInfo` will be innacurate here because the actual page is filtering, but the `getPage` call that `pageInfo` uses does not filter. I think we want to pass in `(pagination: Pagination) => grantService.getPage(pagination, filter)` to `pageInfo` instead.

See https://github.com/interledger/rafiki/blob/main/packages/backend/src/graphql/resolvers/webhooks.ts. 

This is beyond the scope of this task, but just as an aside, I think this a good reason for returning `pageInfo` from `getPage` itself IMO. Instead of deriving it from a separate function call. I believe that was what was suggested here https://github.com/interledger/rafiki/pull/342#discussion_r879841635

File: packages/auth/src/graphql/schema.graphql (hunk 2 above)
Comment (BlairCurrey) [reply packages/auth/src/graphql/schema.graphql:3]: My opinion isn't very firm here TBH. Other than being consistent. I mostly preferred separate because that's how we saw the relay spec implemented elsewhere and it's also less to break/refactor (but I guess that's not a major consideration). I'm still leaning that way for those reasons but I'm wondering what you think @mkurapov because I think all of our initial instincts were to not use separate args.

Comment (mkurapov) [reply packages/auth/src/graphql/schema.graphql:3]: Even though I agree w @dragosp1011 (I like having the object type instead of the listed arguments), let's stay consistent and have the 4 variables instead. We can then do a swoop of all of the changes in one go to convert to object at some point in the future.

File: packages/auth/src/graphql/resolvers/grant.ts (hunk 3 above)
Comment (mkurapov) [reply packages/auth/src/graphql/resolvers/grant.ts:23]: Good catch :+1:

File: packages/auth/src/graphql/resolvers/grant.test.ts [hunk 4]
```diff
@@ -48,6 +50,13 @@ describe('Grant Resolvers', (): void => {
         const grant = await createGrant(deps)
         grants.push(grant)
       }
+
+      const filter = {
+        identifier: {
+          in: ['test']
+        }
+      }
```
Comment (mkurapov) [inline packages/auth/src/graphql/resolvers/grant.test.ts:58]: This isn't being passed into the actual graphql query

File: packages/auth/src/graphql/resolvers/grant.test.ts [hunk 5]
```diff
@@ -84,6 +94,88 @@ describe('Grant Resolvers', (): void => {
     })
   })
 
+  describe('Grant By id Queries', (): void => {
+    let grant: GrantModel
+    beforeEach(async (): Promise<void> => {
+      const createdGrant = await createGrant(deps)
+      grant = (await getGrantById(deps, createdGrant.id)) || createdGrant
```
Comment (mkurapov) [inline packages/auth/src/graphql/resolvers/grant.test.ts:101]: Won't we always have `createdGrant` defined?

File: packages/auth/src/graphql/schema.graphql [hunk 6]
```diff
@@ -95,3 +129,6 @@ enum GrantState {
   "grant was rejected"
   REJECTED
 }
+
+scalar UInt8
+scalar UInt64
```
Comment (mkurapov) [inline packages/auth/src/graphql/schema.graphql:134]: These scalars would need to be added to the resolvers if you would like to use them. See: `packages/backend/src/graphql/resolvers/index.ts`. Otherwise, you can leave adding limits for a separate PR (if necessary)
